lib/SparqlView.js

```diff
@@ -50,7 +52,18 @@ class SparqlView extends ApiObject {
       })
     }
 
-    return this.client.constructQuery(query, options).then((result) => {
+    return Promise.resolve().then(() => {
+      // TODO: very ugly hack. maybe we can define different rdf types for update and query SPARQL code
```

Comment:
What about distinguishing by file name in the loader? https://www.w3.org/TR/sparql11-query/ says

> It is recommended that sparql query files have the extension ".rq" (lowercase) on all platforms.

While on https://www.w3.org/TR/sparql11-update/ it reads

> It is recommended that SPARQL Update files have the extension ".ru" (lowercase) on all platforms.